server/mscalendar/filters.go

```diff
@@ -41,16 +43,18 @@ func withClient(m *mscalendar) error {
 	if m.client != nil {
 		return nil
 	}
-	client, err := m.MakeClient()
-	if err != nil {
-		return err
-	}
 
+	var client remote.Client
+	var err error
 	if m.actingUser.MattermostUserID == m.Config.BotUserID {
```

Comment:
It is still necessary this BotUserID check here?

Reply:
Yes, if we are in a function that calls `withClient`, we want the bot user to be able to create its client without explicitly calling `withSuperuserClient`. This if statement takes care of that.